pkg/controller/prowjob/prowjob-controller.go

```diff
@@ -0,0 +1,149 @@
+
+	"github.com/openshift/release-controller/pkg/controller"
+)
+
+const (
+	// prowJobControllerAgentName is the controller name for the ProwJob controller
+	prowJobControllerAgentName = "release-payload-prowjob-controller"
+
+	// SuccessSynced is used as part of the Event 'reason' when a ProwJob has synced
+	SuccessSynced = "Synced"
+
+	// MessageResourceSynced is the message used for an Event fired when a ProwJob is synced successfully
+	MessageResourceSynced = "ProwJob synced successfully"
+)
+
+// ProwJobController is the controller implementation for ProwJob resources
+type ProwJobController struct {
+	controller.Controller
+	prowJobLister lister.ProwJobLister
+	// ctx is the passed-down global context. It's used and passed
+	// down to all API client calls as to make sure all in-flight calls get
+	// cancelled if the main context is
+	ctx context.Context
+	// controllerCancel is the components global cancelFunc. It's used to
+	// cancel the global context, stop the leader election and subsequently
+	// initiate a shut down of all control loops
+	controllerCancel context.CancelFunc
+	// Recorder is an event recorder for recording Event resources to the
+	// Kubernetes API.
+	recorder record.EventRecorder
+}
+
+// NewProwJobController returns a new ProwJob controller
+func NewProwJobController(
+	controllerContext context.Context,
+	controllerCancel context.CancelFunc,
+	kubeClientset kubernetes.Interface,
+	prowJobInformer informer.ProwJobInformer,
+	prowJobNamespace string) *controller.Controller {
+
+	utilruntime.Must(prowscheme.AddToScheme(scheme.Scheme))
+
+	// Create event broadcaster
+	klog.V(4).Info("Creating event broadcaster")
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartStructuredLogging(0)
+	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{Interface: kubeClientset.CoreV1().Events(prowJobNamespace)})
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: prowJobControllerAgentName})
+
+	prowJobController := &ProwJobController{
+		prowJobLister:    prowJobInformer.Lister(),
+		ctx:              controllerContext,
+		controllerCancel: controllerCancel,
+		recorder:         recorder,
+	}
+
+	c := controller.NewController(
+		[]cache.InformerSynced{prowJobInformer.Informer().HasSynced},
+		prowJobController,
+		prowJobControllerAgentName,
+		reflect.TypeOf(&v1.ProwJob{}),
+	)
+
+	prowJobFilter := func(obj interface{}) bool {
+		if prowJob, ok := obj.(*v1.ProwJob); ok {
+			if _, ok := prowJob.Labels["release.openshift.io/verify"]; ok {
+				return true
+			}
+		}
+		if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+			if prowJob, ok := tombstone.Obj.(*v1.ProwJob); ok {
+				if _, ok = prowJob.Labels["release.openshift.io/verify"]; ok {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	prowJobInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: prowJobFilter,
+		Handler: cache.ResourceEventHandlerFuncs{
```

Comment:
This handler needs to  use the prowjob to look up the release status that it needs to requeue and should queue the releasestatus key *not* the prowjob key.